### ISSUE TITLE
WCAG 2.2 Focus more visible - can a visible change only in size be a failure?

### ISSUE THREAD

**Issue by eschafer** [w3c/wcag#812]
I was looking through the WCAG 2.2 working process and came across @alastc's draft for more visible focus indicators. Thanks for the work everyone is doing on this!

I was wondering - would it make sense to have a rule that says a visible change only in size would not count as a valid focus indicator?

There's an example in this draft that shows gov.uk's focus indicator on a search box:
![search box in default state (white text field on dark blue background) and focus state (yellow outline added outside the text field)](https://user-images.githubusercontent.com/1136332/60603465-cdc49580-9d83-11e9-8bfe-7ed35b06f58a.png)

The text field gets a yellow outline around it, which doesn't have enough contrast against the white text field but does against the dark blue background. This is considered valid because the outline is outside the text field, so a visible change in contrast did occur.

But it's effectively the same as making the textbox slightly bigger since some people would just see the yellow outline as part of the white box:
![search box in default state (white text field on dark blue background) and focus state (text field slightly bigger)](https://user-images.githubusercontent.com/1136332/60605049-ceaaf680-9d86-11e9-8933-99bc5697b693.png)

I'm concerned that this would mean that you could just make an element bigger by 1px on each side and have it pass (the outline in the example above is 3px, so it could be even less noticeable). I've seen this before where buttons grow a tiny amount when they're focused. At best you'd be able to maybe notice this change as it happens, but it'd be almost impossible to notice if you're coming back to a page without knowing where the focus is.

**Comment by alastc** [comment]
> would it make sense to have a rule that says a visible change only in size would not count as a valid focus indicator?

I created a [test page](https://alastairc.uk/tests/wcag22-examples/focus-more-visible.html) to evaluate a few examples against different methods of testing:

- Adjacent colors (either), which doesn't catch some very difficult to see examples (3, 8, 12), I think that should be discounted as an approach.
- Adjacent colors (both).
- Change of color.

I think that avoiding the possibility of 'change of size' would involve the indicator having contrast on **_both_** sides, such as examples 1, 4, 7, 13.

That does discount a lot of (apparently) very obvious indicators, such as inverting colors and thicker borders (examples 2 & 5).

Whilst I don't disagree with your point, I would be very hesitant to 'ban' examples like the yellow outer border which are well tested.

**Comment by eschafer** [comment]
This is just from me poking around so it might not be the right thing to look at, but it looks like gov.uk is changing their focus styles and I think this is a preview of their current progress:
https://govuk-frontend-review.herokuapp.com/

**Comment by alastc** [comment]
Yep, Gov.uk has been working on those updates for a while. They do try to go above and beyond the minimum (guidelines) which is great.

**Comment by patrickhlauke** [comment]
> I'm concerned that this would mean that you could just make an element bigger by 1px on each side and have it pass

similar to the original problem of "focus visible" (where it wasn't normatively defined what "visible" means, in measurable terms), this would end up again creating potentially arbitrary restrictions/thresholds of what does and does not constitute "more visible". there's a danger once again that you either have extremes of "following the normative wording, but not actually making it (more) visible in any practical sense", or you just very unscientifically start trying to define stuff (for instance, i wouldn't say banning just a size change outright is the right approach...imagine controls that grow to, say, double their size when focused...that would be an extremely "more visible" case that would be nonsensical to outlaw)

**Comment by alastc** [comment]
We have a fairly robust (if complex) set of three rules within the new criteria:
https://www.w3.org/TR/WCAG22/#focus-visible-enhanced

I think that addresses this issue?
